moped-editor/src/views/projects/projectView/MilestoneTemplateModal.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect, useMemo } from "react";
+import {
+  Button,
+  Box,
+  Checkbox,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  IconButton,
+  List,
+  ListItemText,
+  ListItemIcon,
+  ListItem,
+  TextField,
+} from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { makeStyles } from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import AddCircle from "@material-ui/icons/AddCircle";
+import { returnSignalPHBMilestoneTemplate } from "../../../utils/timelineTemplates";
+
+import { ADD_PROJECT_MILESTONE } from "../../../queries/project";
+import { useMutation } from "@apollo/client";
+
+const useStyles = makeStyles((theme) => ({
+  dialogTitle: {
+    color: theme.palette.primary.main,
+    fontFamily: theme.typography.fontFamily,
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+}));
+
+const templateChoices = ["AMD", "PDD"];
+
+ /**
+  * useMemo hook to choose milestone options
+  * @return {Object[]}
+  */
+const useMilestoneOptions = (template, projectId) =>
+  useMemo(() => {
+    if (template === "AMD") {
+      return returnSignalPHBMilestoneTemplate(projectId);
+      // } else if (template === "PDD") {
+      //   // etc return [];
+    } else {
+      return [];
+    }
+  }, [template, projectId]);
+
+const MilestoneTemplateModal = ({
+  isDialogOpen,
+  handleDialogClose,
+  projectId,
+  milestoneNameLookup,
+  selectedMilestones,
+  refetch,
+}) => {
+  const classes = useStyles();
+
+  const [template, setTemplate] = useState(null);
+  const [milestonesToAdd, setMilestonesToAdd] = useState([]);
+
+  const [addProjectMilestone] = useMutation(ADD_PROJECT_MILESTONE);
+
+  // Array of milestone ids already in moped_proj_milestones
+  const selectedMilestonesIds = selectedMilestones.map(
+    (milestone) => milestone.milestone_id
+  );
+
+  // milestone options with existing milestones filtered out
+  const milestonesList = useMilestoneOptions(template, projectId).filter(
+    (option) => !selectedMilestonesIds.includes(option.milestone_id)
+  );
```

Comment:
@chiaberry the hook looks good, but when i run locally the useEffect is firing on repeat and checkboxes aren't changing state.

i think you want to pass `selectedMilestones` into the `useMilestoneOptions` hook and do all your filtering there. alternatively, you could make a separate memo that filters the `milestonesList` to excluded the preexisting milestones.